test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -159,6 +159,48 @@ void performFailoverTest() {
         log.info("Read count: {}, write count: {} with Segment Store  and controller failover after sleep", currentReadCount1, currentWriteCount1);
     }
 
+    void performFailoverForTestsInvolvingTxns() {
+
+        log.info("Test with 3 controller, segment store instances running and without a failover scenario");
+        log.info("Read count: {}, write count: {} without any failover",
+                testState.eventReadCount.get(), testState.eventWriteCount.get());
+
+        //check reads and writes after sleeps
+        log.info("Sleeping for {} ", WAIT_AFTER_FAILOVER_MILLIS);
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
+        log.info("Read count: {}, write count: {} without any failover after sleep before scaling",
+                testState.eventReadCount.get(),  testState.eventWriteCount.get());
+
+        //Scale down segment store instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        //zookeeper will take about 30 seconds to detect that the node has gone down
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
+        log.info("Scaling down Segment Store instances from 3 to 2");
+        log.info("Read count: {}, write count: {} after Segment Store  failover after sleep",
+                testState.eventReadCount.get(),  testState.eventWriteCount.get());
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 30 seconds to detect that the node has gone down
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
+        log.info("Scaling down controller instances from 3 to 2");
+        log.info("Read count: {}, write count: {} after controller failover after sleep",
```

Comment:
We could remove this log since we are not using the read/ write count for verification.